modules/tensor_mechanics/doc/content/documentation/systems/Postprocessors/MaterialTimeStepPostprocessor.md

```diff
@@ -1,19 +1,55 @@
-<!-- MOOSE Documentation Stub: Remove this when content is added. -->
+# Material Time Step Postprocessor
 
-# MaterialTimeStepPostprocessor
+!syntax description /Postprocessors/MaterialTimeStepPostprocessor
 
-!alert construction title=Undocumented Class
-The MaterialTimeStepPostprocessor has not been documented, if you would like to contribute to MOOSE by
-writing documentation, please see [/generate.md]. The content contained on this page explains
-the typical documentation associated with a MooseObject; however, what is contained is ultimately
-determined by what is necessary to make the documentation clear for users.
+## Description
 
-!syntax description /Postprocessors/MaterialTimeStepPostprocessor
+The `MaterialTimeStepPostprocessor` works in conjunction with inelastic material
+models to compute the appropriate maximum time step for the allowable inelastic
+strain increment; the user specifies this maximum strain increment in the inelastic
+material model.
+The maximum time step size is a numerical tool used to ensure the calculation of
+physically reasonable inelastic strains and to improve convergence of the inelastic
+material model.
+\begin{equation}
+  \label{eqn:limiting_ts}
+  \Delta t |_{limit} = \Delta t \cdot \frac{\Delta \epsilon^{inel}_{max}}{\Delta \epsilon^{inel}}
+\end{equation}
+where $\Delta \epsilon^{inel}_{max}$ is the maximum effective inelastic strain
+increment, set by the user, and $\Delta \epsilon^{inel}$ is the current scalar
+effective inelastic strain increment.
+
+The `MaterialTimeStepPostprocessor` collects the time step limitations from all
+of the quadrature points in the simulation mesh and stores the minimum value.
+This minimum allowable time step size value is then used by the
+[IterationAdaptiveDT](/Executioner/TimeStepper/IterationAdaptiveDT.md)
+to restrict the time step size based on the limit calculated in the previous
+time step.
```

Comment:
I think it should be mentioned that when using the MaterialTimeStepPostprocessor with IterationAdaptiveDT that the value from the postprocessor is only used if it is less than what the IterationAdaptiveDT would normally use for the time step.